client/src/hooks.ts

```diff
@@ -18,7 +18,7 @@ import { isValidLocale } from "../../libs/locale-utils";
 
 export function useLocale() {
   const { locale } = useParams();
-  return isValidLocale(locale) ? locale : DEFAULT_LOCALE;
+  return isValidLocaleMC(locale) ? locale : DEFAULT_LOCALE;
```

Comment:
Maybe we could call a `parseLocale()` method here instead that returns `en-US` for `en-us`, and `null` for invalid locales? Otherwise this `useLocale()` hook would now return `en-US` if locale is `zh-cn`.